src/actions/clusterActions.js

```diff
@@ -40,79 +40,145 @@ function computeCapabilities(cluster, provider) {
   return capabilities;
 }
 
+// This is a helper function, not an action creator.
+// For some reason we are storing clusters as objects instead as arrays
```

Comment:
For convenience I use arrays instead objects whenever possible, eg when I have to store a bunch of objects of the same shape (instantiations of the same class) because it's easier to manage them, but objects are ok also.

Another of the refactors I have in mind is related to the shape of the store, but I don't know if it's worth it and really needed. So we will stick to what we have now, which works perfectly fine